apps/files_sharing/lib/SharingBlacklist.php

```
@@ -88,10 +88,27 @@ private function initCache() {
 		}
 	}
 
-	private function fetchBlacklistedReceiverGroupIds() {
+	/**
+	 * @param bool $returnAnArrayWithGroupsAsValues if true then the group ids are the values in the array
+	 *                                              if false then the group ids are the values in the array
+	 * @return string[] the list of group ids
+	 */
+	private function fetchBlacklistedReceiverGroupIds($returnAnArrayWithGroupsAsValues = false) {
 		$configuredBlacklist = $this->config->getAppValue('files_sharing', 'blacklisted_receiver_groups', '[]');
 		$decodedGroups = \json_decode($configuredBlacklist, true);
-		// expected a plain array here
+
+		if (!\is_array($decodedGroups)) {
+			$decodedGroups = [];
+			$jsonError = \json_last_error();
+			\OC::$server->getLogger()->error('SharingBlacklist::getBlacklistedReceiverGroups - can\'t decode blacklisted_receiver_groups: ' . \print_r($configuredBlacklist, true) . ' - json error code: ' . $jsonError, ['app' => 'files_sharing']);
```

just ignore the error message. This needs to be injected in order to include unittests, which is too much work for little benefit. Adding a comment in the PHPDocs saying that an empty list will be returned if we can't parse the json is enough.
Note that there is no need for anyone to know that we're storing the information as a json string.